dbt/adapters/trino/connections.py

```diff
@@ -435,9 +441,13 @@ def open(cls, connection):
         return connection
 
     @classmethod
-    def get_response(cls, cursor) -> AdapterResponse:
+    def get_response(cls, cursor) -> TrinoAdapterResponse:
         message = "SUCCESS"
-        return AdapterResponse(_message=message)
+        return TrinoAdapterResponse(
+            _message=message,
+            query_id=cursor._cursor.query_id,
+            compiled_query=cursor._cursor._query._sql,
```

Comment:
This should probably come from a public property from the cursor. We'll have to add this in Trino Python client (not blocking for this PR).